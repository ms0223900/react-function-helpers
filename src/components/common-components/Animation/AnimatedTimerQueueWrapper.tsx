import React, { ReactNodeArray } from 'react';
import AnimationWrapper, { AnimationWrapperWithoutChildrenProps } from './AnimationWrapper';
import TimerQueueWrapper from './TimerQueueWrapper';

export interface AnimatedTimerQueueWrapperProps extends AnimationWrapperWithoutChildrenProps {
  intervalSeconds?: number
  children: ReactNodeArray
}

export interface IntervalAndAnimationTimes {
  intervalSeconds?: number,
  animationDuration?: number
}

export const getDefaultAnimationDurationFromIntervalSeconds = ({ intervalSeconds, animationDuration }: IntervalAndAnimationTimes) => (
  typeof animationDuration === 'number' ? animationDuration : intervalSeconds
);

const AnimatedTimerQueueWrapper = (props: AnimatedTimerQueueWrapperProps) => {
  const animationDuration = getDefaultAnimationDurationFromIntervalSeconds(props);
  const Children = React.Children.map(props.children, (child) => (
    <AnimationWrapper 
      {...props}
      animationDuration={animationDuration}
    >
      {child}
    </AnimationWrapper>
  ));

  return (
    <TimerQueueWrapper {...props}>
      {Children}
    </TimerQueueWrapper>
  );
};

export default AnimatedTimerQueueWrapper;