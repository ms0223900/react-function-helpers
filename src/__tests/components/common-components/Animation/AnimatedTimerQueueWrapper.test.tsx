import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import AnimatedTimerQueueWrapper, {
  getDefaultAnimationDurationFromIntervalSeconds
} from 'components/common-components/Animation/AnimatedTimerQueueWrapper';

describe('test AnimatedTimerQueueWrapper wrapper', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <AnimatedTimerQueueWrapper>
        <></>
        <></>
      </AnimatedTimerQueueWrapper>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test getDefaultAnimationDurationFromIntervalSeconds function', () => {
    const animationDuration = 10;
    const props = {
      intervalSeconds: 1,
    };
    const defaultFromIntervalSeconds = getDefaultAnimationDurationFromIntervalSeconds(props);
    expect(defaultFromIntervalSeconds).toEqual(props.intervalSeconds);

    const _animationDuration = getDefaultAnimationDurationFromIntervalSeconds({
      ...props,
      animationDuration,
    });
    expect(_animationDuration).toEqual(animationDuration);
  });
});