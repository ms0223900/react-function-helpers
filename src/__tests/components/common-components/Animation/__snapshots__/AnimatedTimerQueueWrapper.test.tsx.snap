// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AnimatedTimerQueueWrapper wrapper should render correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 slideInDown"
  style={
    Object {
      "animationDelay": "0s",
      "animationDuration": "1s",
      "animationFillMode": "forwards",
    }
  }
/>
`;
